src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNetCoreMvc2Integration.cs

```diff
@@ -63,22 +95,22 @@ public AspNetCoreMvc2Integration(object actionDescriptorObj, object httpContextO
             TargetAssembly = "Microsoft.AspNetCore.Mvc.Core",
             TargetType = "Microsoft.AspNetCore.Mvc.Internal.MvcCoreDiagnosticSourceExtensions")]
         public static void BeforeAction(
-                                        object diagnosticSource,
-                                        object actionDescriptor,
-                                        dynamic httpContext,
-                                        object routeData)
+            object diagnosticSource,
+            object actionDescriptor,
+            object httpContext,
+            object routeData)
         {
             AspNetCoreMvc2Integration integration = null;
 
             try
             {
                 integration = new AspNetCoreMvc2Integration(actionDescriptor, httpContext);
-                IDictionary<object, object> contextItems = httpContext.Items;
+                IDictionary<object, object> contextItems = ((dynamic)httpContext).Items;
```

Comment:
This looks like it can be converted to use TryGetPropertyValue to get rid of the cast.

Reply:
Die, `dynamic`, die.